wasmi_v1/src/arena/mod.rs

```diff
@@ -202,11 +202,11 @@ where
     }
 }
 
-/// An iterator over exlusive references of arena entities and their indices.
+/// An iterator over exclusive references of arena entities and their indices.
 #[derive(Debug)]
 pub struct IterMut<'a, Idx, T> {
     iter: iter::Enumerate<slice::IterMut<'a, T>>,
-    __marker: PhantomData<fn() -> Idx>,
+    __marker: PhantomData<Idx>,
```

Comment:
This change would change semantic in a way that is not useful for a non-owning `IterMut`. Please revert. Read more here: https://doc.rust-lang.org/nomicon/phantom-data.html

Reply:
ok

Reply:
But TBH, I thinke `slice::Iter<'a, T>` has holded the correct lifetime semantic.

Reply:
Otherwise, Should write in this way?
```rust
#[derive(Debug)]
pub struct Iter<'a, Idx, T> {
    iter: iter::Enumerate<slice::Iter<'a, T>>,
    __marker: PhantomData<&'a Idx>,
}
```

Reply:
The compiler do not report any error and there are no `unsafe code`, So I think both of them should be correct.

Reply:
While I agree that `PhantomData<&'a Idx>` is generally probably a good idea for iterators I kinda have to disagree for this particular case since we are not iterating over `&Idx` but over `Idx` by value in the iterator. Therefore I'd suggest using `fn() -> Idx` as before this PR.

Reply:
fine

Reply:
This change request is still outstanding before we can merge. Rest looks good to me.